### ISSUE TITLE
AttributeValue still coupled to product (i think)

### ISSUE THREAD

**Issue by peteward** [Sylius/Sylius#3984]
`AttributeController::renderAttributesAction` decouples quite nicely to allow any attribute type, but it then renders the following template:

```
return $this->render('SyliusAttributeBundle::attributeChoice.html.twig', array('form' => $form->createView()));
```

This uses a hard-coded route:

```
 <form action="{{ path('sylius_backend_render_attribute_forms') }}" method="GET" id="attributeChoice">
```

Which is set to the product attribute controller:
```
sylius_backend_render_attribute_forms:
    path: /attribute-forms
    methods: [GET]
    defaults:
        _controller: sylius.controller.product_attribute:renderAttributeValueFormsAction
```

Any ideas ?

Will try working around by overriding more of it in the meantime.

**Comment by pjedrzejewski** [comment]
Definitely still coupled. :) But should be easy to fix, we will have a look! Thanks for report!

**Comment by peteward** [comment]
The next one is `dynamic-property-types.js` which embeds new attribute form elements into the page after selecting them and is also hard-coded to products:

```
$(document).ready(function() {
    'use strict';

    $('a[data-collection-button="add"]').on('click', function (e) {
        var collectionContainer = $('#' + $(this).data('collection'));
        var lastElementNumber = (collectionContainer.children().length) - 1;
        $('#sylius_product_properties_' + lastElementNumber + ' .property-chooser').handlePrototypes({
            prototypePrefix: 'property-prototype',
            prototypeElementPrefix: '',
            containerSelector: '#sylius_product_properties_' + lastElementNumber + ' .control-group:last .controls'
        });
        $('#sylius_product_properties_' + lastElementNumber + ' .property-chooser').change(function() {
            $('#sylius_product_properties_' + lastElementNumber + ' .control-group:last .controls input, #sylius_product_properties_' + lastElementNumber + ' .control-group:last .controls select').each(function() {
                this.name = this.name.replace(/__name__/g, lastElementNumber);
                this.id = this.id.replace(/__name__/g, lastElementNumber);
            });
        });
    });
});
```

**Comment by peteward** [comment]
Hmm that wasn't it actually, it was the template called by the above route `attributeValueForms` which is also hard-coded for products.

The workaround was to override both of these methods and replace both of their templates.

These are the templates that are coupled:
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AttributeBundle/Resources/views/attributeValueForms.html.twig
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AttributeBundle/Resources/views/attributeTypes.html.twig
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AttributeBundle/Resources/views/attributeChoice.html.twig

**Comment by pjedrzejewski** [comment]
Fixed in #4014, thanks a lot for reporting Peter!
